### DISCUSSION TITLE
Servidores e Hospedagens

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#50] [Discussão]
Eu sei que esse é um topico clichê na maioria das comunidades, mas eu queria abrir um pra falar de hospedagens e servidores em geral (não vi nenhum até agora), eu to começando a escopar a aprender Node e Ruby, e to pensando em pegar um na Digital Ocean.

Alias, eu pago 1200 reais = 300 dolares na Bluehost por ano no plano PRO pra hospedar apenas 10 sites em wordpress.Compensa aprender Amazon AWS ou Digital Ocean? Eu nao manjo muito dessa parte mas aprendo rápido e me interesso.

Vejo a galera tambem comentando as ferramentas de deploy, que na época que eu trabalhava na Centauro quem mexia era o pessoal de CM.. (Jenkins)

O que um leigo que sempre hospedou seus sites em servidores compartilhados PHP/MySQL/Linux tem que aprender em questão de preço e custo beneficio, pra manjar dos paranaue de hospedagens?

Quem viveu a época do Hypermart,Virtualave,Angelfire,Geocites e dos redirecionadores de url (rg3.net,cjb.net,da.ru)?? 

Alias meu plano na Bluehost vence em Abril,antes disso eu quero pegar um na Digital Ocean ou equivalente (é o que amaioria dos devs tem me indicado), pra começar a transferir meus clientes em wordpress, e tambem ter um pra eu começar a brincar com coisas minhas..

Na verdade meio que sempre foi assim, quando era só PHP eu instalava todos os scripts do mundo,de e-commerce,cms,redes sociais, vasculhava aquele diretório (hotscripts.com) pra por pra testar e fuçar...

Nunca gostei de fazer nada muito do zero, pq pensava que se tinha pronto era so hospedar e rodar, e mexer em cima kkk, mas agora to mudando esse pensamento...

Enfim, se puderem discutir o assunto  ai, valeu!!

**E fazer uma lista dos que você não recomenda**

**Comment by raphaklaus** [comment]
Recomendo [Digital Ocean](www.digitalocean.com). Porque tem planos super baratos (o primeiro começa em 5 dólares mensais), as máquinas virtuais sobem numa velocidade sinistra, tem uma API rica, documentação boa, suporte também.

Tem outros, que dizem que são bons, mas são mais carinhos: [Azure](https://azure.microsoft.com/pt-br/), [AWS](https://aws.amazon.com/pt/)

**Quem viveu a época do Hypermart,Virtualave,Angelfire,Geocites e dos redirecionadores de url (rg3.net,cjb.net,da.ru)??**

Boa época Ahaha! Minha primeira hospedagem foi no VilaBol em 2001 :heart: Depois fui pro PHP em 2004 eu usei o falecido Webcindario.

**O que um leigo que sempre hospedou seus sites em servidores compartilhados PHP/MySQL/Linux tem que aprender em questão de preço e custo beneficio, pra manjar dos paranaue de hospedagens?**

Basicamente:
- Conhecer comandos Linux. Como instalar pacotes, remover arquivos de forma segura, configurar sistema operacional, entre outras operações básicas.
- Geração de chaves públicas, privadas, SSH
- Ferramentas de deploy, tipo [Capistrano](http://capistranorb.com/) (Ruby) ou [ShipIt](https://github.com/shipitjs/shipit) (JS)

---

A mudança da maneira de gerenciar seus sites muda bastante. Se você nunca teve experiência com servidores dedicados, se prepare pois será um bom aprendizado (eu penei no início :see_no_evil:), mas não é um bicho de 7 cabeças. Depois você verá que vale muito mais à pena em todos os sentidos.

Te recomendaria a instalar um [Vagrant](https://www.vagrantup.com/) da vida e ir treinando, pois a forma com que irá lidar com tudo será bem semelhante.

Abraços :smiley:

**Comment by rubenmarcus** [comment]
Valeu, mas SSH é tipo igual é com o GIT ??? e o bom dessas hospedagens é que são escalaveis certo? valeria a pena pegar 1 server pra freelas e 1 pra projetos pessoais?? eu registrei dois dominios ontem de projetos que quero tocar e quero que sejam grandes... grandes eu falo tipo (10000 visitas diarias no minimo) , quando eu fazia sites de animes e musica nenhum site teve mais que 5000, então 10000 diarias pra mim é grande kkkkkkk Não sei se os bluehost aguentam..  os projetos que quero fazer pessoais são sobre guerras e história que é um hobby meu, e pensei porque não aprender minhas deficiencias em front e back escrevendo e publicando sobre assuntos que já leio no dia dia..

**Comment by juniorconte** [comment]
Se vc está familiarizado com terminal, não terá problema algum com SSH, pois é um terminal remoto.
Quanto a escala, Digital Ocean não é auto-scaling, e você deve ficar atento a alguns detalhes:

1 - Diferentemente de AWS e Azure, se vc deixar seu host desligado, vc ainda continua pagando pelo valor / hora. Pois seus recursos continuam alocados e disponíveis para a qualquer momento voltar a ativa.

2 - As máquinas só "escalam" para cima. Como não da pra diminuir o disco, vc só consegue aumentar o tamanho da máquina. Minha sugestão, é que se espaço em disco não for vital para você, crie a menor VM possível, você poderá escalar para cima optando por manter o disco no tamanho reduzido, isso permitira escalar para baixo posteriormente.

3 - Digital Ocean não é um serviço de PaaS, e você precisará de alguns conhecimentos de Sysadmin para deixar as coisas redondas. Embora existam vários "One-Click Applications", nem tudo está disponível dessa forma. Considere utilizar serviços como Bitnami para abstrair a implantação de N stacks diferentes. https://bitnami.com/digitalocean

4 - Mesmo você tendo que meter a mão-na-massa, existem tutoriais para quase tudo na comunidade. Você dificilmente não encontrara o que precisa. https://www.digitalocean.com/community

Eu utilizo Digital Ocean, recomendo fortemente.

**Comment by rickbenetti** [comment]
Aqui na Studio BOZ mudamos da [OVH](https://www.ovh.pt/servidores_dedicados/) (tinhamos dedicado) mas a conexão para Europa é muito lenta e demorava em média 2 a 3 segundos à mais do que se fosse nos EUA ou aqui no Brasil.

Eu testei [Digital Ocean](https://m.do.co/c/dbc47675fc7b) e realmente é muito bom, eles tem tutoriais super práticos pra iniciar as configurações dos projetos, como utilizo uma solução estilo [Bedrock](https://roots.io/bedrock/) para sites WordPress, toda meu deploy ocorre com [Capistrano](https://www.youtube.com/watch?v=PlxrSTtoff8).

Mas acabamos escolhendo  [Linode](https://www.linode.com/?r=e16cb8f87f872ac6e62394b061eec4a7430fd83d), pra gente foi a melhor solução, tudo muito rápido também.

Quanto a questão de custos, reduzimos mais de 50% uma vez que não oferecemos mais acesso à cPanel e nem FTP, todo acesso se limita ao site do site basicamente.

**Comment by tcelestino** [comment]
Digital Ocean é uma boa opção. Não recomendaria para sites WordPress (se for institucional), porque é um custo x trabalho que não vejo a necessidade. Se você tiver criando uma aplicação, que você precisa ter um controle de tudo, realmente é a opção, mas para sites WordPress, minha sugestão é optar por servidores compartilhados (KingHost, Hostgator, etc...).

Vale a pena aprender, porque é "outro mundo".
